poseidon/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+mod parameters;
+
+type CircuitField = ark_bls12_381::Fr;
+type FpVar = ark_r1cs_std::fields::fp::FpVar<CircuitField>;
+
+pub mod hash {
+    use ark_bls12_381::Fr;
+    use ark_std::vec;
+    use poseidon_permutation::Instance;
+
+    use crate::parameters::RATE_1_PARAMETERS;
+    /// hashes one field value, outputs a fixed length field value
+    pub fn one_to_one_hash(domain_separator: &Fr, value: Fr) -> Fr {
```

Comment:
I believe the `domain_separator` is needed mostly on the theoretic side, so I'd just fixed one and remove the parameter here and in the module below.

Reply:
done